d2bs/kolbot/libs/bots/BattleOrders.js

```diff
@@ -42,20 +83,29 @@ function BattleOrders() {
 
 	Pather.moveTo(me.x + 6, me.y + 6);
 
-	var i,
+	let i,
 		tick = getTickCount(),
 		failTimer = 60;
 
 MainLoop:
 	while (true) {
 		switch (Config.BattleOrders.Mode) {
 		case 0: // Give BO
+			if (this.AmTardy()) {
```

Comment:
its a bit silly to do the same check in both cases, might be more worthy to add this check in singular form before the switch?

Reply:
yep, good idea